nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -129,8 +205,39 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         final String cassandraTable = context.getProperty(TABLE).evaluateAttributeExpressions(inputFlowFile).getValue();
         final RecordReaderFactory recordParserFactory = context.getProperty(RECORD_READER_FACTORY).asControllerService(RecordReaderFactory.class);
         final int batchSize = context.getProperty(BATCH_SIZE).evaluateAttributeExpressions().asInteger();
-        final String batchStatementType = context.getProperty(BATCH_STATEMENT_TYPE).getValue();
         final String serialConsistencyLevel = context.getProperty(CONSISTENCY_LEVEL).getValue();
+        final String updateKeys = context.getProperty(UPDATE_KEYS).evaluateAttributeExpressions(inputFlowFile).getValue();
```

Comment:
Checking correct property values could be done in a `customValidate` method.

Reply:
I think you are referring to the update keys/update method being present when the statement type is update. However, I don't think that would work with setting the statement type using an attribute (cql.statement.type). In that case we would still need the current validation logic and I'm a bit hesitant to have it duplicated.

Reply:
this is now implemented.